src/GitVersion.Core/VersionCalculation/NextVersionCalculator.cs

```diff
@@ -146,4 +147,177 @@ private static void EnsureHeadIsNotDetached(GitVersionContext context)
     }
 
     private static bool MajorMinorPatchEqual(SemanticVersion lastTag, SemanticVersion baseVersion) => lastTag.Major == baseVersion.Major && lastTag.Minor == baseVersion.Minor && lastTag.Patch == baseVersion.Patch;
+
+    private NextVersion Calculate(IBranch branch, Config configuration)
```

Comment:
Is it possible to break this method up into several smaller methods, or preferably its own (private) class?

Reply:
This hole method needs to be refactored (not in the PR). I'm not sure if it makes it better if we refactor this in other methods or classes. In my opinion I should look like the following:
```csharp
    private NextVersion Calculate(IBranch branch, Config configuration)
    {
        using (log.IndentLog("Calculating the base versions"))
        {
            var nextVersions = GetNextVersions(branch, configuration).ToArray();
            return nextVersions.Max(element => element.IncrementedVersion);
        }
    }
```
That's why I would suggest not to change it because it was just moved from the BaseVersionCalculator. Is it okay with you to refactor it later?